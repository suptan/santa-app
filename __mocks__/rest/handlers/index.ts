import { santaRests } from "./santa";
import { wishRests } from "./wish";

const restHandlers = [...wishRests, ...santaRests];

export { restHandlers };
