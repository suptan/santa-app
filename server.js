// // server.js
// // where your node app starts

// // init project
// const express = require('express');
// const morgan = require('morgan');
// const path = require('path');
// const app = express();
// const bodyParser = require('body-parser');

// app.use(bodyParser());
// app.use(morgan());

// // we've started you off with Express,
// // but feel free to use whatever libs or frameworks you'd like through `package.json`.

// app.use(express.static(path.join(__dirname, '../react-app/build')));
// // http://expressjs.com/en/starter/static-files.html
// app.use(express.static('public'));

// // http://expressjs.com/en/starter/basic-routing.html
// app.get('/', (request, response) => {
//   response.sendFile(__dirname + '/views/index.html');
// });

// // listen for requests :)
// // const listener = app.listen(process.env.PORT || 3000, function () {
// //   console.log('Your app is listening on port ' + listener.address().port);
// // });
