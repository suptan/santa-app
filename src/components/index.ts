export * from "./Button";
export * from "./Form";
export * from "./Footer";
export * from "./Header";
export * from "./Input";
export * from "./Loader";
export * from "./Wish";
