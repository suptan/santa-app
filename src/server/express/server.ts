import express, { Request, Response } from "express";
import path from "path";
import morgan from "morgan";
import Router from "express-promise-router";
import * as dotenv from 'dotenv'
import { initializeUser, initializeUserProfile } from "./user";
import { wishRouter } from "./routes/wish";
dotenv.config();
const port = process.env.SERVER_PORT || 9001;
const app = express();
const apiRouter = Router();

// (async () => {
//   if (process.env.NODE_ENV === "test") {
//     console.log(window);
//     await window.__mockServerReady;
//   }
// })();

app.use(morgan("tiny"));
app.use(express.static("./dist"));
app.use(express.json());

// app.get('/', (request, response) => {
//   response.sendFile(__dirname + '/views/index.html');
// });
// apiRouter.use((_, __, next) => {
//   console.log("api routes");
//   next();
// });

apiRouter.get("/health", (_: Request, res: Response) => {
  const status = {
    uptime: process.uptime(),
    message: "OK",
    timestamp: Date.now(),
  };

  try {
    res.send(status);
  } catch (e: any) {
    status.message = e;
    res.status(503).send("Down");
  }
});

app.use("/api", apiRouter, wishRouter);

// Skip server listener when execute in test environment
if (process.env.NODE_ENV !== "test") {
  initializeUser();
  initializeUserProfile();

  app.listen(port, function () {
    console.log(`Your app is listening on port ${port}`);
  });
}

export { app };
