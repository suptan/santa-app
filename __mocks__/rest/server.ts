import { setupServer } from "msw/node";
import { restHandlers } from "./handlers";

const server = setupServer(...restHandlers);

export { server }