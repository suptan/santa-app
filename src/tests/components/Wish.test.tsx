import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { http, HttpResponse } from "msw";
import toast from "react-hot-toast";
import { Wish } from "@/components";
import { server } from "@/../__mocks__/rest/server";

jest.mock("react-hot-toast", () => ({
  __esModule: true,
  default: {
    success: jest.fn(),
    error: jest.fn(),
  },
}));

describe("<Wish>", () => {
  const usernamePlaceholder = "Good boys and girls";
  const wishPlaceholder = "What do you want for Christmas?";
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it("should render with blank form", () => {
    const { container } = render(<Wish />);
    expect(container).toMatchSnapshot();
  });

  it("should display success message", async () => {
    render(<Wish />);
    await userEvent.type(
      screen.getByPlaceholderText(usernamePlaceholder),
      "username1"
    );
    await userEvent.type(
      screen.getByPlaceholderText(wishPlaceholder),
      "Mr. Santa, I wish for the happiness."
    );
    await userEvent.click(screen.getByText("Submit"));
    expect(toast.success).toHaveBeenCalledWith("Your wish will be fulfilled.");
  }, 15000);

  it("should display unauthorized error", async () => {
    server.use(
      http.post(
        "/api/wish",
        () => {
          throw new HttpResponse(null, { status: 401 });
        },
        { once: true }
      )
    );
    render(<Wish />);
    await userEvent.type(
      screen.getByPlaceholderText(usernamePlaceholder),
      "rnadom"
    );
    await userEvent.type(screen.getByPlaceholderText(wishPlaceholder), "none");
    await userEvent.click(screen.getByText("Submit"));
    expect(toast.error).toHaveBeenCalledWith("Unauthorized");
  }, 15000);
});
