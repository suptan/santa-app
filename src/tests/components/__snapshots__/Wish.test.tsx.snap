// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wish> should render with blank form 1`] = `
<div>
  <section>
    <h2
      class="text-4xl font-bold mb-4 dark:text-white"
    >
      Make a wish to Santa
    </h2>
    <form
      class="mb-4 flex flex-col gap-2 border rounded p-2"
    >
      <div
        class="flex flex-col gap-1 text-gray-400"
      >
        <label
          for="name"
        >
          ID:
        </label>
        <div
          class="border border-zinc-200 rounded bg-transparent"
        >
          <input
            aria-labelledby="name"
            class="bg-transparent p-2 w-full"
            id="name"
            maxlength="50"
            name="name"
            placeholder="Good boys and girls"
            value=""
          />
        </div>
      </div>
      <div
        class="flex flex-col gap-1 text-gray-400"
      >
        <label
          for="msg"
        >
          Message:
        </label>
        <div
          class="border border-zinc-200 rounded bg-transparent"
        >
          <textarea
            aria-labelledby="msg"
            class="bg-transparent p-2 w-full"
            id="msg"
            name="msg"
            placeholder="What do you want for Christmas?"
            type="textarea"
          />
        </div>
      </div>
      <section
        class="flex gap-2"
      >
        <button
          class="py-2 px-4 rounded [&[disabled]]:bg-slate-200 bg-red-600 text-white hover:bg-red-500"
          type="button"
        >
          Clear
        </button>
        <button
          class="py-2 px-4 rounded [&[disabled]]:bg-slate-200 bg-green-600 text-white hover:bg-green-500"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </section>
    </form>
    <div
      class="flex flex-col gap-1 text-gray-400"
    >
      <label>
        one
      </label>
      <div
        class="border border-zinc-200 rounded bg-transparent"
      >
        <input
          class="bg-transparent p-2 w-full"
          value=""
        />
      </div>
    </div>
  </section>
</div>
`;
